controller/routes.go

```diff
@@ -0,0 +1,44 @@
+package controller
+
+import (
+	"github.com/Baraulia/COURIER_SERVICE/middleware"
+	"github.com/Baraulia/COURIER_SERVICE/model"
+	"github.com/gin-gonic/gin"
+	"github.com/gorilla/mux"
+)
+
+type Handler struct {
+	services *model.Service
+}
+
+func NewHandler(services *model.Service) *Handler {
+	return &Handler{services: services}
+}
+
+func (h *Handler) InitRoutes() *mux.Router {
+	r := mux.NewRouter()
+	//r.HandleFunc("/orders/completed", h.GetCourierCompletedOrders).Methods("GET")
+	//r.HandleFunc("/orders", h.GetAllOrdersOfCourierService).Methods("GET")
+	//r.HandleFunc("/orders/bymonth",h.GetCourierCompletedOrdersByMonth).Methods("GET")
+	//r.HandleFunc("/orders",h.AssigningOrderToCourier).Methods("PUT")
+	return r
+}
+
+
+func (h *Handler) InitRoutesGin() *gin.Engine {
+	router := gin.Default()
+
+	router.Use(
+		middleware.CorsMiddleware,
+	)
+
+	order := router.Group("/orders")
+	{
+		order.GET("/completed", h.GetCourierCompletedOrders)
+		order.GET("/", h.GetAllOrdersOfCourierService)
+		order.GET("/bymonth",h.GetCourierCompletedOrdersByMonth)
+		order.PUT("/", h.AssigningOrderToCourier)
```

Comment:
The final url is `PUT /orders`, so from REST convention it means only that some order is updated, not specifically that it assigns order to courier. So it is better to have more generic name for the controller function. But inside controller it is ok if you limit possible updates only to updating courier and for all other updates return `not implemented` or `not-allowed` or so on

Reply:
fixed, thanks a lot!